### ISSUE TITLE
Add keyword `remove_destination` to `mv` function

### ISSUE THREAD

**Issue by peter1000** [JuliaLang/julia#11145]
This is a follow up issue from comments:
https://github.com/JuliaLang/julia/pull/11024#issuecomment-99271305
> julia could add the kwarg: `remove_destination::Bool=false` to **mv** to be in line with [cp](http://julia.readthedocs.org/en/latest/stdlib/file/?highlight=cp#Base.cp)
>
> ```
> cp(src::AbstractString, dst::AbstractString; remove_destination::Bool=false, follow_symlinks::Bool=false)
> ```
>
> If that is wanted I could do it.

[reply StefanKarpinski](https://github.com/JuliaLang/julia/pull/11024#issuecomment-99272053)
@StefanKarpinski > Yes please, @peter1000!    

---

I just had a fast look and I think there are some other issues too:

1. current docs say `Move a file from src to dst`.   Related to second question: is it suppose to work also on directories?

2. The real issue is here again similar we had in the `cp` discussion. symlinks. I tried a fast `mv` on a directory and obviously any absolute symlinks within the rootdir get broken if the folder is renamed.

I can fix this but the question is: should we also add the kw: `follow_symlinks` similar to copy.

```
cp(src::AbstractString, dst::AbstractString; remove_destination::Bool=false, follow_symlinks::Bool=false)
```

Relevant code portion: [fs.jl](https://github.com/JuliaLang/julia/blob/master/base/fs.jl#L116)
[file.jl](https://github.com/JuliaLang/julia/blob/master/base/file.jl#L118)

**Comment by peter1000** [comment]
according to my reading of the src: `mv` with the current implementation was only supposed for files: [uv_fs_rename lbuv](http://docs.libuv.org/en/latest/fs.html?highlight=uv_fs_rename#c.uv_fs_rename) 
```
Equivalent to rename(2).
```

[rename(2)](http://linux.die.net/man/2/rename)

```
Name
rename - change the name or location of a file
Synopsis
#include <stdio.h>

int rename(const char *oldpath, const char *newpath);
Description
rename() renames a file, moving it between directories if required.
```

**Comment by simonster** [comment]
`remove_destination` seems reasonable, but I don't think we should try to do anything about absolute symlinks when moving files. I don't know of anything else that tries to handle this case.

`rename` should work just fine for directories. From the man page you linked:

> oldpath can specify a directory. In this case, newpath must either not exist, or it must specify an empty directory.

But I noticed that `mv` does not presently perform a recursive copy when the `rename` throws an error (i.e., when the source and destination are on different file systems). It seems like it should.

**Comment by peter1000** [comment]
just something I noticed: currently the test suit uses: `Base.samefile` to test drectories stat info. [src](https://github.com/JuliaLang/julia/blob/master/test/file.jl#L100)

I just wondered if the name `Base.samefile` is not a bit misleading in such a case. Would it be worth a consideration to have a `samedir` or just a `samepath`.

**Comment by simonster** [comment]
`samefile` is the Python name. I think having separate functions for files and directories would be annoying and not very useful. `samepath` doesn't seem like it would do the same thing since the same file can reside at multiple paths due to hard links. I think `sameinode` would be more correct (except that NTFS doesn't call them inodes) but "inode" may be too technical for us.

**Comment by StefanKarpinski** [comment]
I think sticking with the Python name here is probably best and making sure the documentation is clear.

**Comment by peter1000** [comment]
There is no official documentation as it is not exported: I could add a small comment that it is supposed to work for files and directories when I send this PR.

**Comment by jakebolewski** [comment]
closed by #11172
